src/hera/workflow_template.py

```diff
@@ -0,0 +1,176 @@
+"""The implementation of a Hera workflow for Argo-based workflows"""
```

Comment:
What do you think about specifically mentioning workflow templates rather than just workflows?